src/post/post.repository.knex.ts

```diff
@@ -152,4 +154,92 @@ export class KnexPostRepository {
             throw err;
         }
     }
+
+    async updateReplyCount(postId: number, increment = true): Promise<void> {
+        const transaction = await this.db.transaction();
+
+        try {
+            await transaction('posts')
+                .where({ id: postId })
+                .update({
+                    reply_count: this.db.raw(
+                        `reply_count ${increment ? '+' : '-'} 1`,
+                    ),
+                });
+
+            await transaction.commit();
+        } catch (err) {
+            await transaction.rollback();
+            throw err;
+        }
+    }
+
+    async getById(id: number): Promise<Post | null> {
+        const row = await this.db('posts')
+            .join('accounts', 'accounts.id', 'posts.author_id')
+            .where('posts.id', id)
+            .select(
+                'posts.id',
+                'posts.uuid',
+                'posts.type',
+                'posts.audience',
+                'posts.title',
+                'posts.excerpt',
+                'posts.content',
+                'posts.url',
+                'posts.image_url',
+                'posts.published_at',
+                'posts.like_count',
+                'posts.repost_count',
+                'posts.reply_count',
+                'posts.reading_time_minutes',
+                'posts.author_id',
+                'posts.ap_id',
+                'posts.in_reply_to',
+                'posts.thread_root',
+                'accounts.username',
+                'accounts.name',
+                'accounts.bio',
+                'accounts.avatar_url',
+                'accounts.banner_image_url',
+            )
+            .first();
+
+        if (!row) {
+            return null;
+        }
+
+        const author = new Account(
+            row.author_id,
+            row.username,
+            row.name,
+            row.bio,
+            parseURL(row.avatar_url),
+            parseURL(row.banner_image_url),
+            null,
+        );
+
+        const post = new Post(
```

Comment:
I don't like doing this. Maybe there is a better way to map the row with post entity. 
We might not want to create a new Post object every time we write a function to fetch a row from posts table. This will be even more messy when we'll have to get multiple rows.